core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -142,10 +142,17 @@ public void available() {
                 final boolean isSystemDatabase = db.databaseName().equals(GraphDatabaseSettings.SYSTEM_DATABASE_NAME);
                 Configuration config = dependencyResolver.resolveDependency(ApocConfig.class).getConfig();
                 var initializers = collectInitializers(isSystemDatabase, config);
+
+                // Wait for database to become available, once it is available
+                // procedures are also available by necessity.
                 if (!isSystemDatabase) {
                     awaitApocProceduresRegistered();
-                } else if (!initializers.isEmpty()) {
-                    waitForSystemDbLifecycles();
+                } else {
+                    if (!initializers.isEmpty()) {
+                        waitForSystemDbLifecycles();
+                        // Register listener to drop apoc nodes from system database once a database is deleted
+                        databaseEventListeners.registerDatabaseEventListener(new SystemFunctionalityListener());
+                    }
```

Comment:
I was thinking inside the else but not inside the `if (!initializers.isEmpty())`? Or did you have a logical reason for why it is inside the if?